src/app/core/market/api/template/template.model.ts

```diff
@@ -74,8 +74,24 @@ export class Template {
     return this.object.ItemInformation.ItemImages;
   }
 
+  get intPrice(): any {
+    const itemPrice = this.getPrice();
+    if (itemPrice) {
+      return new Amount(itemPrice.basePrice).getIntegerPart();
+    }
+    return undefined;
+  }
+
+  get centsPrice(): any {
+    const itemPrice = this.getPrice();
+    if (itemPrice) {
+      return new Amount(itemPrice.basePrice).getFractionalPart();
+    }
+    return undefined;
+  }
+
   setBasePrice(): void {
-    const itemPrice = this.object.PaymentInformation.ItemPrice;
+    const itemPrice = this.getPrice();
     if (itemPrice) {
       this.basePrice = new Amount(itemPrice.basePrice);
```

Comment:
lets set the price once, as an `Amount` object instead of having to instantiate the object twice